### ISSUE TITLE
When PSPs are enabled, csi-registrar container crashes due to read-only filesystem 

### ISSUE THREAD

**Issue by funkypenguin** [topolvm/topolvm#398]
When deploying topolvm-node pods via the helm chart 3.1.2, the csi-registrar container crashes with the following error:

```
I1119 02:32:51.961824       1 main.go:164] Version:
I1119 02:32:51.961876       1 main.go:165] Running node-driver-registrar in mode=registration
I1119 02:32:51.966324       1 node_register.go:52] Starting Registration Server at: /registration/topolvm.cybozu.com-reg.sock
I1119 02:32:51.966798       1 node_register.go:61] Registration Server started at: /registration/topolvm.cybozu.com-reg.sock
I1119 02:32:51.966923       1 node_register.go:91] Skipping healthz server because HTTP endpoint is set to: ""
I1119 02:32:51.977493       1 main.go:100] Received GetInfo call: &InfoRequest{}
E1119 02:32:51.977676       1 main.go:105] "Failed to create registration probe file" err="mkdir /var/lib/kubelet: read-only file system" registrationProbePath="/var/lib/kubelet/plugins/topolvm.cybozu.com/node/registration"
I1119 02:32:51.977736       1 main.go:107] "Kubelet registration probe created" path="/var/lib/kubelet/plugins/topolvm.cybozu.com/node/registration"
I1119 02:32:52.021812       1 main.go:118] Received NotifyRegistrationStatus call: &RegistrationStatus{PluginRegistered:true,Error:,}
```

I've tracked this down to the fact that the [PSP deployed by the helm chart](https://github.com/topolvm/topolvm/blob/main/charts/topolvm/templates/node/psp.yaml#L29) sets `readOnlyRootFilesystem: true`

As a result, my topolvm-node pods are continuously crashlooping.

**To Reproduce**

1. Use a PSP-enabled cluster
2. Deploy topolvm using the helm chart, and set `podSecurityPolicy.create: true`
3. Observe the topolvm-node pods crashlooping
4. Observe the logs from the csi-registrar containers (`kubectl logs -n topolvm-system -l app.kubernetes.io/name=topolvm-node -c csi-registrar`)
5. Edit the PSP (`kubectl edit psp topolvm-node`) and delete the `readOnlyRootFilesystem` line
6. Restart the topolvm-node daemonset rollout (`kubectl rollout restart -n topolvm-system daemonset topolvm-node`)
7. Observe that the pods no longer crashloop, the error about read-only mode is gone 🥳 

**Expected behavior**

We didn't **used** to need read/write mode in the PSPs. If we need it now because of changes/improvements to the CSI registrar, then the PSP should be updated accordingly.

**Additional context**

I'm happy to submit a PR to fix the PSP, but I don't know enough about CSI to understand _why_ the registrar container needs write access to `/var/lib/kubelet`. If a developer will point me in the right direction, I'll prepare a PR.

D

**Comment by toshipp** [comment]
I enabled livenessProbe for csi-node-driver-registrar on https://github.com/topolvm/topolvm/pull/374, which makes a file to check registration and if a directory for the file does not exist, csi-node-driver-registrar makes the directory.
I could not find the problem because PSP is now deprecated and it is not well tested.
The solution is to prepare an emptyDir and mount it to /var/lib/kubelet/plugins/topolvm.cybozu.com/node/.
If you provide a PR, I'm happy to review it.

**Comment by toshipp** [comment]
closed via #410
